### PR TITLE
allow samesite in cookies

### PR DESCRIPTION
**Pull request by samuelcolvin**
## What do these changes do?

Allow `samesite` to be allowed in cookeis, see also:
* https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
* https://www.chromestatus.com/feature/5088147346030592

## Are there changes in behavior for the user?

no, just new `samesite=None` keyword argument to `set_cookie()`

## Related issue number

* #2502 partial
* https://github.com/aio-libs/aiohttp-session/issues/376

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by asvetlov** [comment]
Hmm.
I'm biased; should we monkey-patch Python<3.8 or restrict the feature to Python 3.8+ only?

Guys, what are your opinions?

**Comment by samuelcolvin** [comment]
I'm happy with either, just let me know and I'll update the PR.

**Comment by samuelcolvin** [comment]
But given that chrome showing a warning for any cookie not including `SameSite`, I imagine many people will want this fairly soon - restricting it to python >= 3.8 will cause headaches for lots of people.

File: aiohttp/web_response.py [hunk 1]
```diff
@@ -42,6 +42,12 @@
     BaseClass = collections.abc.MutableMapping
 
 
+if 'samesite' not in Morsel._reserved:  # type: ignore
```
Comment (asvetlov) [inline aiohttp/web_response.py]: Use `helpers.PY_38` check here instead if `Morsel._reserved` lookup.
It makes the intention explicitly clear and allows dropping the patch after dropping Python 3.7 support.

**Comment by samuelcolvin** [comment]
Done, with `versionadded`, docs look like:
image

is that okay?

File: docs/web_reference.rst [hunk 2]
```diff
@@ -702,6 +703,14 @@ StreamResponse
                           specification the cookie
                           conforms. (Optional, *version=1* by default)
 
+      .. versionadded:: 3.7
```
Comment (asvetlov) [inline docs/web_reference.rst]: Please move the directive inside `:param:` block, see `web.Response` `zlib_executor_size` for inspiration.

**Comment by asvetlov** [comment]
@samuelcolvin ping

**Comment by samuelcolvin** [comment]
done, sorry for delay.

**Comment by samuelcolvin** [comment]
:ok_hand:

**Comment by AndreiPashkin** [comment]
@asvetlov, I wonder - is this going to be released anytime soon?

**Comment by bitnom** [comment]
Very important security consideration. Should consider a new release just to get this feature out the door imo. I'm tempted to install from master. Any commits in master I should be afraid of at this point?

**Comment by vishim** [comment]
@asvetlov  - Is there anything blocking the 3.7 release? Chrome has already started rolling out the samesite update to a selected population.

**Comment by asvetlov** [comment]
This pull request has been mentioned on **aio-libs**. There might be relevant details there:

https://aio-libs.discourse.group/t/patch-samesite-cookies-to-minor-version/111/1

**Pull request merged**